src/pkg/packager/deploy.go

```diff
@@ -451,7 +454,9 @@ func (p *Packager) setupState(ctx context.Context) (err error) {
 		// Try to create the zarf namespace
 		spinner.Updatef("Creating the Zarf namespace")
 		zarfNamespace := cluster.NewZarfManagedNamespace(cluster.ZarfNamespaceName)
-		if _, err := p.cluster.CreateNamespace(ctx, zarfNamespace); err != nil {
+		// NOTE: Should this ignore if already exists?
+		_, err := p.cluster.Clientset.CoreV1().Namespaces().Create(ctx, zarfNamespace, metav1.CreateOptions{})
```

Comment:
Original implementation skips create if namespace already exists. Is this required here?

Reply:
It's not necessary to check if the namespace exists before attempting to create it. However, we should handle the case where the `zarf` namespace already exists.

We can use the [`k8s.io/apimachinery/pkg/api/errors`](https://pkg.go.dev/k8s.io/apimachinery@v0.29.1/pkg/api/errors#pkg-overview) package to check if the error indicates that the namespace already exists:

```go
_, err := p.cluster.Clientset.CoreV1().Namespaces().Create(ctx, zarfNamespace, metav1.CreateOptions{})
if err != nil && !kerrors.IsAlreadyExists(err) {
	spinner.Fatalf(err, "Unable to create the zarf namespace")
}
```

This will fix the current failure in the `DevDeploy` test

Reply:
Ok that is good to know that it is actually required. In that case this should be a update or create function to make sure that any metadata changes are applied.